src/frontend/features/user/userSlice.js

```diff
@@ -0,0 +1,94 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+import axios from "axios";
+
+const initialState = {
+	allusers: [],
+	user: null,
+	userPosts: [],
+	userStatus: "idle",
+	error: "",
+};
+
+const getAllUsers = createAsyncThunk("/user/getAllUsers", async () => {
+	const { data } = await axios.get("/api/users");
+	return data;
```

Comment:
it's good if you use try catch here in side the createAsyncThunk and user and inside the catch 
return thunkAPI.rejectWithValue(error);

Reply:
I don't need to catch the error message here so I am not using try catch here